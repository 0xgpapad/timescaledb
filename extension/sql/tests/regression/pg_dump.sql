\set ON_ERROR_STOP 1
\set VERBOSITY verbose
\set SHOW_CONTEXT never

\o /dev/null
\ir include/insert.sql
\o

\set ECHO ALL


\c postgres

\! pg_dump -h localhost -U postgres -Fc Test1 > dump/Test1.sql
\! pg_dump -h localhost -U postgres -Fc test2 > dump/test2.sql
\! pg_dump -h localhost -U postgres -Fc meta  > dump/meta.sql

\! dropdb -h localhost -U postgres Test1
\! dropdb -h localhost -U postgres test2
\! dropdb -h localhost -U postgres meta

\! pg_restore -h localhost -U postgres -d postgres -C dump/Test1.sql
\! pg_restore -h localhost -U postgres -d postgres -C dump/test2.sql
\! pg_restore -h localhost -U postgres -d postgres -C dump/meta.sql

\c test2
SELECT * FROM "testNs";

--query for the extension tables/sequences that will not be dumped by pg_dump (should be empty)
SELECT objid::regclass, *
FROM pg_catalog.pg_depend
WHERE   refclassid = 'pg_catalog.pg_extension'::pg_catalog.regclass AND 
        refobjid = (select oid from pg_extension where extname='iobeamdb') AND 
        deptype = 'e' AND 
        classid='pg_catalog.pg_class'::pg_catalog.regclass
        AND objid NOT IN (select unnest(extconfig) from pg_extension where extname='iobeamdb')




