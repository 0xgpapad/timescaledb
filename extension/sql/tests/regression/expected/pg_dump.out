psql:include/create_clustered_db.sql:12: NOTICE:  00000: installing required extension "dblink"
LOCATION:  CreateExtensionInternal, extension.c:1441
psql:include/create_clustered_db.sql:12: NOTICE:  00000: installing required extension "postgres_fdw"
LOCATION:  CreateExtensionInternal, extension.c:1441
psql:include/create_clustered_db.sql:12: NOTICE:  00000: installing required extension "hstore"
LOCATION:  CreateExtensionInternal, extension.c:1441
psql:include/create_clustered_db.sql:16: NOTICE:  00000: installing required extension "dblink"
LOCATION:  CreateExtensionInternal, extension.c:1441
psql:include/create_clustered_db.sql:16: NOTICE:  00000: installing required extension "postgres_fdw"
LOCATION:  CreateExtensionInternal, extension.c:1441
psql:include/create_clustered_db.sql:16: NOTICE:  00000: installing required extension "hstore"
LOCATION:  CreateExtensionInternal, extension.c:1441
psql:include/create_clustered_db.sql:20: NOTICE:  00000: installing required extension "dblink"
LOCATION:  CreateExtensionInternal, extension.c:1441
psql:include/create_clustered_db.sql:20: NOTICE:  00000: installing required extension "postgres_fdw"
LOCATION:  CreateExtensionInternal, extension.c:1441
psql:include/create_clustered_db.sql:20: NOTICE:  00000: installing required extension "hstore"
LOCATION:  CreateExtensionInternal, extension.c:1441
\c postgres
\! pg_dump -h localhost -U postgres -Fc Test1 > dump/Test1.sql
\! pg_dump -h localhost -U postgres -Fc test2 > dump/test2.sql
\! pg_dump -h localhost -U postgres -Fc meta  > dump/meta.sql
\! dropdb -h localhost -U postgres Test1
\! dropdb -h localhost -U postgres test2
\! dropdb -h localhost -U postgres meta
\! pg_restore -h localhost -U postgres -d postgres -C dump/Test1.sql
\! pg_restore -h localhost -U postgres -d postgres -C dump/test2.sql
\! pg_restore -h localhost -U postgres -d postgres -C dump/meta.sql
\c test2
SELECT * FROM "testNs";
     timeCustom      | device_id | series_0 | series_1 | series_2 | series_bool 
---------------------+-----------+----------+----------+----------+-------------
 1257894000000000000 | dev1      |      1.5 |        1 |        2 | t
 1257894000000000000 | dev1      |      1.5 |        2 |          | 
 1257894000000001000 | dev1      |      2.5 |        3 |          | 
 1257894001000000000 | dev1      |      3.5 |        4 |          | 
 1257897600000000000 | dev1      |      4.5 |        5 |          | f
 1257894002000000000 | dev1      |      2.5 |        3 |          | 
 1257987600000000000 | dev1      |      1.5 |        1 |          | 
 1257987600000000000 | dev1      |      1.5 |        2 |          | 
 1257894000000000000 | dev20     |      1.5 |        1 |          | 
 1257894000000000000 | dev20     |      1.5 |        2 |          | 
(10 rows)

--query for the extension tables/sequences that will not be dumped by pg_dump (should be empty)
SELECT objid::regclass, *
FROM pg_catalog.pg_depend
WHERE   refclassid = 'pg_catalog.pg_extension'::pg_catalog.regclass AND 
        refobjid = (select oid from pg_extension where extname='iobeamdb') AND 
        deptype = 'e' AND 
        classid='pg_catalog.pg_class'::pg_catalog.regclass
        AND objid NOT IN (select unnest(extconfig) from pg_extension where extname='iobeamdb')
 objid | classid | objid | objsubid | refclassid | refobjid | refobjsubid | deptype 
-------+---------+-------+----------+------------+----------+-------------+---------
(0 rows)

